File: python/lc290.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        toks = s.split(' ')
        if len(pattern)!= len(toks):
            return False
        
        m1 = {}
        m2 = {}
        for i in range(len(pattern)):
            if pattern[i] in m1:
                if m1[pattern[i]] != toks[i]:
                    return False
            else:
                if toks[i] in m2:
                    return False
                m1[pattern[i]] = toks[i]
                m2[toks[i]] = pattern[i]
        return True

File: cs/lc290.cs
public class Solution {
    public bool WordPattern(string pattern, string s) {
        var m1 = new Dictionary<char,string>();
        var m2 = new Dictionary<string,char>();
        string []toks = s.Split(" ");
        if (pattern.Length != toks.Length)
            return false;
        for(int i=0;i<pattern.Length;i++)
        {
            if( m1.ContainsKey(pattern[i]))
            {
                if (m1[pattern[i]] != toks[i])
                    return false;
            }
            else
            {
                if(m2.ContainsKey(toks[i]))
                    return false;
                m1.Add(pattern[i], toks[i]);
                m2.Add(toks[i], pattern[i]);
            }
        }
        return true;
    }
}
